File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        result = []

        def backtrack(start, state, currentSum): 
            if currentSum > target or start == len(nums):
                return 

            if currentSum == target: 
                result.append(state[:]) 
                return

            for i in range(start, len(nums)):
                candidate = nums[i]
                state.append(candidate)

                backtrack(i, state, currentSum + candidate)
                
                state.pop()

        backtrack(0, [], 0)

        return result

File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const charMap = {};

        for (let i = 0; i < nums.length; i++) {
            const a = target - nums[i];
            if (a in charMap) {
                return [i, charMap[a]];
            }

                        charMap[nums[i]] = i;

        }

        return [];
    }
}
